### ISSUE TITLE
Upgrade to lodash 4

### ISSUE THREAD

**Issue by manubo** [AmpersandJS/ampersand-collection-lodash-mixin#10]
Dependency on lodash 3 breaks `require('lodash.merge')` in newer packages (e.g. material-ui) also depending on `lodash-merge`.

**Comment by dhritzkiv** [comment]
Curious that this breaks seeing as how `lodash.merge` and `lodash/merge` are separate modules. Or even if `lodash.merge@v4` was required by the mentioned "newer packages", this module would still load `lodash.merge@v3`, since it would be installed in its `node_modules` directory. Your dependency tree should look like:

``` tree
- lodash.merge@v4
- ampersand-collection-lodash-mixin
| - lodash.merge@v3
```

Could it be something else? What's the specific error?

**Comment by manubo** [comment]
I have an app using `material-ui` and `ampersand`, both having a dependency on `lodash.rest` (versions 4 and 3 respectively).
I have an error when calling `getMuiTheme` of `material-ui`, which requires `lodash.merge@4` but fails to do so because it ends up calling `lodash.rest@3`. I can solve the problem by explicitly adding `lodash.rest@4` as a dependency to my project. 
I was trying to reproduce the issue in a separate project, but failed to do so. As the two packages are the only ones using `lodash.rest`, I blamed `ampersand-collection-lodash-mixin` for using the minor version. Sorry for that. I agree, this cannot be the cause of the error. I can close the issue if you agree?

**Comment by dhritzkiv** [comment]
When npm\* installs dependencies, it _should_ install distinct major versions of that dependency, like so:

``` tree
- material-ui
- lodash.merge@v4.x (used by material-ui)
- ampersand-collection-lodash-mixin
| - lodash.merge@v3.x (lower major version is installed in its dependent's node_modules dir)
```

If you haven't already, it can help to run these from the project root, which should make sure the dependencies are in their right place.
- `rm -rf node_modules`
- `npm install`

---
- npm@3, at least. npm@2 had different behaviour, but I can't remember in which way it was different

**Comment by lukekarrys** [comment]
Closed by #11
